docroot/sites/classrooms.uiowa.edu/modules/classrooms_core/classrooms_core.module

```diff
@@ -594,3 +643,93 @@ function classrooms_core_preprocess_views_view(&$variables) {
+  $view = $variables['view'];
+  $variables['row_item_classes'] = [];
+  $display_block = $view->current_display;
+
+  $style = $view->style_plugin;
+  $rows = $variables['rows'];
+
+  switch ($display_block) {
+    case 'block_rooms':
+
+      $attribute_classes = [
+        'card--centered',
+        'card--stacked',
+        'card--alignment-center',
+        'card--enclosed',
+        'media--widescreen',
+      ];
+
+      $variables['row_item_classes'] = array_merge($attribute_classes, array_values($variables['row_item_classes']));
+      foreach ($rows as $id => $row) {
+        // Set remaining row_item_classes as attributes on each row.
+        $variables['rows'][$id]['attributes']->addClass($variables['row_item_classes']);
+
+        // Get node path to link card with.
+        $nid = $view->result[$id]->_entity->id();
+        $link = \Drupal::service('path_alias.manager')->getAliasByPath('/node/' . $nid);
+
+        // Combine fields to display.
+        $text_combined = $style->getField($id, 'field_room_type') . "<br />" . $style->getField($id, 'field_room_max_occupancy');
+        $text = ['#markup' => $text_combined];
+        $title_combined = $style->getField($id, 'field_room_room_id') . ' ' . strtoupper($view->result[$id]->node__field_room_building_id_field_room_building_id_target_i);
+        $title = ['#markup' => $title_combined];
+
+        // Initialize card_options variable.
+        $variables['rows'][$id]['card_options'] = [
+          'attributes' => $variables['rows'][$id]['attributes'],
+          'media_sizes' => ['card__media--large'],
+          'card_image' => $style->getField($id, 'field_image'),
+          'card_title' => \Drupal::service('renderer')->render($title),
+          'card_link_url' => $link,
+          'card_link_title' => 'Room details',
+          'card_text' => \Drupal::service('renderer')->render($text),
+          'visible_button' => TRUE,
+        ];
+      }
+      break;
+
+    case 'block_programmed_rooms_list':
+
+      $attribute_classes = [
+        'card--left',
+        'card--stacked',
+        'card--alignment-left',
+        'card--enclosed',
+      ];
+
+      $variables['row_item_classes'] = array_merge($attribute_classes, array_values($variables['row_item_classes']));
+      foreach ($rows as $id => $row) {
+        // Set remaining row_item_classes as attributes on each row.
+        $variables['rows'][$id]['attributes']->addClass($variables['row_item_classes']);
+
+        // Get node path to link card with.
+        $nid = $view->result[$id]->_entity->id();
+        $link = \Drupal::service('path_alias.manager')->getAliasByPath('/node/' . $nid);
+
+        // Combine fields to display.
+        $text_combined = '<strong>Room name:</strong> ' . $style->getField($id, 'field_room_name') . "<br /><strong>Capacity:</strong> " . $style->getField($id, 'field_room_max_occupancy') . "<br /><strong>Responsible Unit:</strong> " . $style->getField($id, 'field_room_responsible_unit');
```

Comment:
untested, but i wonder if there are a couple places where the following could be used to generate regular field markup instead of crafting custom labels and adding spacing:

```php
$style->getField($id, 'field_room_name')->view(['label' => 'inline', 'type' => 'string']);

or

$style->getField($id, 'field_room_building_id')->view(['label' => 'visually_hidden', 'type' => 'entity_reference_entity_id']); 
```

Reply:
Worth trying! It seems like there has to be a simpler way than all the custom stuff.